opslevel/resource_opslevel_check_repository_grep.go

```diff
@@ -0,0 +1,102 @@
+package opslevel
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	opslevel "github.com/opslevel/opslevel-go/v2022"
+)
+
+func resourceCheckRepositoryGrep() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a repository file check",
+		Create:      wrap(resourceCheckRepositoryGrepCreate),
+		Read:        wrap(resourceCheckRepositoryGrepRead),
+		Update:      wrap(resourceCheckRepositoryGrepUpdate),
+		Delete:      wrap(resourceCheckDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: getCheckSchema(map[string]*schema.Schema{
+			"directory_search": {
+				Type:        schema.TypeBool,
+				Description: "Whether the check looks for the existence of a directory instead of a file. Defaults to false",
+				ForceNew:    false,
+				Optional:    true,
+			},
+			"filepaths": {
+				Type:        schema.TypeList,
+				MinItems:    1,
+				Description: "Restrict the search to certain file paths.",
+				ForceNew:    false,
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"file_contents_predicate": getPredicateInputSchema(false),
+		}),
+	}
+}
+
+func resourceCheckRepositoryGrepCreate(d *schema.ResourceData, client *opslevel.Client) error {
+	input := opslevel.CheckRepositoryGrepCreateInput{}
+	setCheckCreateInput(d, &input)
+
+	input.DirectorySearch = d.Get("directory_search").(bool)
+	input.Filepaths = getStringArray(d, "filepaths")
+	input.FileContentsPredicate = expandPredicate(d, "file_contents_predicate")
+
+	resource, err := client.CreateCheckRepositoryGrep(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(resource.Id.(string))
+
+	return resourceCheckRepositoryGrepRead(d, client)
+}
+
+func resourceCheckRepositoryGrepRead(d *schema.ResourceData, client *opslevel.Client) error {
+	id := d.Id()
+
+	resource, err := client.GetCheck(id)
+	if err != nil {
+		return err
+	}
+
+	if err := setCheckData(d, resource); err != nil {
+		return err
+	}
+	if err := d.Set("directory_search", resource.RepositoryGrepCheckFragment.DirectorySearch); err != nil {
+		return err
+	}
+	if err := d.Set("filepaths", resource.RepositoryGrepCheckFragment.Filepaths); err != nil {
```

Comment:
this is an array - do we need to do anything special here to "flatten"?  Do we store arrays in other usecases?

Reply:
Yeah, we're doing the same in RepoFile check.  But now that you mention it, not sure if that's a bad assumption to make that it's working correctly there.  Didn't see anything weird in the state while testing it out.